### PR TITLE
Update various styles to better align with BCGov Design

### PR DESCRIPTION
**Pull request by jujaga**
# Description

This PR resolves a large number of various style fixes to better align with BCGov Design principles. This is needed in order to ensure accessibility requirements are met.

- Vertically center the request account form button texts
- Link text underline should only show on hover
- Home page card tweaks
- My Applications page - card rounding
- Revisit read-only visual indications on read-only idir and email fields
- Confirm if RequestForm submit button be disabled until form is valid
- Request Account page spacing
- Button icons have underlines when hovered

[SHOWCASE-881](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-881)

## Types of changes

Bug fix (non-breaking change which fixes an issue)
New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-k8vopl-tools.pathfinder.gov.bc.ca/dashboard?id=nr-get-token-pr-161

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-k8vopl-tools.pathfinder.gov.bc.ca/dashboard?id=nr-get-token-pr-161

File: app/frontend/src/assets/scss/style.scss [hunk 1]
```diff
@@ -10,14 +24,17 @@ h3 { font-size: 1.444em; /* (25.92px) */ }
 h4 { font-size: 1.2em; /* (21.60px) */ }
 
 h1, h2, h3, h4, h5 {
+  color: #494949;
```
Comment (loneil) [inline app/frontend/src/assets/scss/style.scss:27]: Once we expand on our theming knowledge we can make stuff like #494949 SASS variables (could do in the existing file anyways, but would leave it till we determine how we want to set Vuetify SASS detaults in a theme)

File: app/frontend/src/assets/scss/style.scss [hunk 2]
```diff
@@ -28,47 +45,32 @@ a {
   }
 }
 
+// General Button Style Extensions
+.v-btn {
+  span > span {
```
Comment (loneil) [inline app/frontend/src/assets/scss/style.scss:50]: Does this need to be specified to span > span? Or can we just say font weight is bold inside buttons in general?

File: app/frontend/src/assets/scss/style.scss [hunk 3]
```diff
@@ -28,47 +45,32 @@ a {
...
+    span > span {
+      text-decoration: underline;
+    }
+  }
 
-.v-btn.v-btn--disabled {
-  background-color: #eee;
-  color: #777;
-}
+  &.v-btn--disabled {
```
Comment (loneil) [inline app/frontend/src/assets/scss/style.scss:61]: I don't know if you need the & here. If this is trying to select  v-btn.v-btn--disabled I think just having the nesting accomplishes that...

**Pull request merged**